src/providers/Video/Video.provider.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { createContext, useContext, useState } from 'react';
+
+import data from '../../data/youtube-videos-mock.json';
+
+const VideoContext = createContext({
+  query: '',
+  videos: [],
+  selectedVideo: {},
+});
+
+function useVideos() {
+  const context = useContext(VideoContext);
+  if (!context) {
+    throw new Error(`Can't use "useVideos" without an VideoProvider!`);
+  }
+  return context;
+}
+
+function VideoProvider({ children }) {
+  const [query, setQuery] = useState('');
+  const [videos, setVideos] = useState(data);
+  const [selectedVideo, setSelectedVideo] = useState({});
+
+  return (
+    <VideoContext.Provider
+      value={{
+        videos,
+        setVideos,
+        selectedVideo,
+        setSelectedVideo,
+        query,
+        setQuery,
+      }}
+    >
+      {children}
+    </VideoContext.Provider>
+  );
+}
```

Comment:
this is a good use of the Context API and hooks; the only thing I would add it's the actual fetching of the videos and just export the search function in values so you can use it instead of manually define it on the video list. This approach is good enough though.

Reply:
I agree, I think I need some help. I tried creating an async function in Video.provider.jsx and also in Video.reducer.jsx in PR5 but did not work, I could not get the code wait for the api call response